pkg/controller/common/certificates/ca_reconcile.go

```diff
@@ -232,6 +235,66 @@ func internalSecretForCA(
 	}, nil
 }
 
+func detectCAFileNames(path string) (string, string, error) {
+	files := map[string]bool{
+		CertFileName:  false,
+		KeyFileName:   false,
+		CAFileName:    false,
+		CAKeyFileName: false,
+	}
+	for f := range files {
+		exists, err := fs.FileExists(filepath.Join(path, f))
+		if err != nil {
+			return "", "", err
+		}
+		files[f] = exists
+	}
+	switch {
+	case (files[CertFileName] || files[KeyFileName]) && files[CAKeyFileName]:
+		return "", "", fmt.Errorf("both tls.* and ca.* files exist, configuration error")
+	case files[CAFileName] && files[CAKeyFileName]:
+		return filepath.Join(path, CAFileName), filepath.Join(CAKeyFileName), nil
+	case files[CertFileName] && files[KeyFileName]:
+		return filepath.Join(path, CertFileName), filepath.Join(path, KeyFileName), nil
+	}
+	return "", "", fmt.Errorf("no CA certificate files found: %+v", files)
```

Comment:
I was trying to think how we can avoid to leak the map implementation in the error message, and also exit early if the directory does not even exist.

```suggestion
func detectCAFileNames(path string) (string, string, error) {
	if !fs.FolderExists(path) {
		return "", "", fmt.Errorf("global CA directory %s does not exist", path)
	}
	caFiles := []string{CAFileName, CAKeyFileName}
	tlsFiles := []string{CertFileName, KeyFileName}
	existsInDirectory := map[string]bool{}

	for _, f := range append(caFiles, tlsFiles...) {
		exists, err := fs.FileExists(filepath.Join(path, f))
		if err != nil {
			return "", "", err
		}
		existsInDirectory[f] = exists
	}
	switch {
	case (existsInDirectory[CertFileName] || existsInDirectory[KeyFileName]) && existsInDirectory[CAKeyFileName]:
		return "", "", fmt.Errorf("both tls.* and ca.* files exist, configuration error")
	case existsInDirectory[CAFileName] && existsInDirectory[CAKeyFileName]:
		return filepath.Join(path, CAFileName), filepath.Join(CAKeyFileName), nil
	case existsInDirectory[CertFileName] && existsInDirectory[KeyFileName]:
		return filepath.Join(path, CertFileName), filepath.Join(path, KeyFileName), nil
	}
	return "", "",
		fmt.Errorf(
			"no CA certificate files found in %s, expecting one of the following key pair: (%s) or (%s)",
			path,
			strings.Join(caFiles, ","),
			strings.Join(tlsFiles, ","))
}
```

Reply:
Good idea! Do you have an opinion on https://github.com/elastic/cloud-on-k8s/pull/5539#issuecomment-1087481662 i.e. whether we should support two naming conventions for the files to begin with?

Reply:
> Do you have an opinion on [#5539 (comment)](https://github.com/elastic/cloud-on-k8s/pull/5539#issuecomment-1087481662) i.e. whether we should support two naming conventions for the files to begin with?

It seems fair to support both, and I don't think I have a strong argument against this proposal.